### ISSUE TITLE
move all linux e2e (except those using service catalog) tests to OpenShift CI

### ISSUE THREAD

**Issue by kadel** [openshift/odo#1431]
Now when we have working OpenShift CI setup we should move all our e2e runs from TravisCI and CircleCi to OpenShift CI.

All test should run against OpenShift 4.0

Documentation on how to setup clusters for Prow jobs https://github.com/openshift/release/blob/master/ci-operator/README.md#end-to-end-tests

The end result should be that no tests are running on travis-ci or circleci

- [x] e2e_test

- [x] component_test

- [x] java_test

- [x] json_test

- [ ] link_test

- [x] login_test

- [ ] service_test

- [x] source_test

- [ ] watch_test

**Comment by kadel** [comment]
bumping priority as this is getting more and more critical.

**Comment by amitkrout** [comment]
Waiting for https://github.com/redhat-developer/odo/pull/1441 and https://github.com/openshift/release/pull/3031 to merge into master cc_ @mohammedzee1000

**Comment by mohammedzee1000** [comment]
> Waiting for #1441 and [openshift/release#3031](https://github.com/openshift/release/pull/3031) to merge into master cc_ @mohammedzee1000

I am working with @amitkrout on setting up the base expectations for running e2e on OpenShift CI already in above PRS. Once that is done, it will simply be a matter of copy-paste-edit and more PRs to openshift/relasse repo

**Comment by amitkrout** [comment]
I have done some e2e test run on top of pr #1441 with OpenShift 4.0 cluster (on AWS account). 

```make test-json-format-output``` - All scenario passed
```make test-odo-login-e2e``` - All scenario passed


```make test-main-e2e``` - Reported 6 failures (Can be a debug item)

However test like ```test-source-e2e``` and ```test-java-e2e``` are blocked because as per ```odo catalog list components``` seems wildfly is not listed as a supported component types in OpenShift cluster 4.0.  
```
$ odo catalog list components
NAME       PROJECT       TAGS
dotnet     openshift     1.0,1.1,2.1,2.2,latest
httpd      openshift     2.4,latest
java       openshift     8,latest
nginx      openshift     1.10,1.12,1.8,latest
nodejs     openshift     10,6,8,8-RHOAR,latest
perl       openshift     5.24,5.26,latest
php        openshift     7.0,7.1,7.2,latest
python     openshift     2.7,3.5,3.6,latest
ruby       openshift     2.3,2.4,2.5,latest
```

And service catalog is not enabled in the 4.0 cluster causing blocking of test-service-e2e test-watch-e2e and test-link-e2e. I am not sure how service catalog and wildfly component types will be the part of prow setup.

@kadel @mohammedzee1000 Please share your thoughts

**Comment by mohammedzee1000** [comment]
> I have done some e2e test run on top of pr #1441 with OpenShift 4.0 cluster (on AWS account).
> 
> `make test-json-format-output` - All scenario passed
> `make test-odo-login-e2e` - All scenario passed

Awesome we should be able to port these tests over soon enough
> 
> `make test-main-e2e` - Reported 6 failures (Can be a debug item)
> 
> However test like `test-source-e2e` and `test-java-e2e` are blocked because as per `odo catalog list components` seems wildfly is not listed as a supported component types in OpenShift cluster 4.0.
> 
We will have to look at how this can be enabled on the cluster then, maybe?
> ```
> $ odo catalog list components
> NAME       PROJECT       TAGS
> dotnet     openshift     1.0,1.1,2.1,2.2,latest
> httpd      openshift     2.4,latest
> java       openshift     8,latest
> nginx      openshift     1.10,1.12,1.8,latest
> nodejs     openshift     10,6,8,8-RHOAR,latest
> perl       openshift     5.24,5.26,latest
> php        openshift     7.0,7.1,7.2,latest
> python     openshift     2.7,3.5,3.6,latest
> ruby       openshift     2.3,2.4,2.5,latest
> ```
> And service catalog is not enabled in the 4.0 cluster causes blocking of `test-service-e2e` `test-watch-e2e` and `test-link-e2e`. I am not sure how service catalog and wildfly component types will be the part of prow setup.
This is something we will have to look into to see if we can get openshift ci black magic to help us.

If not, we should keep steps handy for how these can be added once a cluster is up
> 
> ```
> $ odo catalog list services
>  ✗  unable to list services because Service Catalog is not enabled in your cluster: unable to get cluster serviceClassExternalName: unable to get cluster service classes: unable to list cluster service classes: clusterserviceclasses.servicecatalog.k8s.io is forbidden: User "developer" cannot list resource "clusterserviceclasses" in API group "servicecatalog.k8s.io" at the cluster scope
> ```
> @kadel @mohammedzee1000 Please share your thoughts

**Comment by amitkrout** [comment]
> > I have done some e2e test run on top of pr #1441 with OpenShift 4.0 cluster (on AWS account).
> > `make test-json-format-output` - All scenario passed
> > `make test-odo-login-e2e` - All scenario passed
> 
> Awesome we should be able to port these tests over soon enough
Sounds good
> 
> > `make test-main-e2e` - Reported 6 failures (Can be a debug item)
> > However test like `test-source-e2e` and `test-java-e2e` are blocked because as per `odo catalog list components` seems wildfly is not listed as a supported component types in OpenShift cluster 4.0.
> 
> We will have to look at how this can be enabled on the cluster then, maybe?
As discussed and tested, the temporary workaround  - ```oc apply -f https://raw.githubusercontent.com/openshift/library/master/arch/x86_64/community/wildfly/imagestreams/wildfly-centos7.json -n openshift``` can be added to the OpenShift ci script with a proper comment

Also the tag ```<TargetFramework>``` - https://github.com/redhat-developer/odo/blob/master/tests/e2e/examples/source/dotnet/asp-net-hello-world.csproj#L4 need to be updated to 2.2 in a separate pr for prow setup.
> 
> > ```
> > $ odo catalog list components
> > NAME       PROJECT       TAGS
> > dotnet     openshift     1.0,1.1,2.1,2.2,latest
> > httpd      openshift     2.4,latest
> > java       openshift     8,latest
> > nginx      openshift     1.10,1.12,1.8,latest
> > nodejs     openshift     10,6,8,8-RHOAR,latest
> > perl       openshift     5.24,5.26,latest
> > php        openshift     7.0,7.1,7.2,latest
> > python     openshift     2.7,3.5,3.6,latest
> > ruby       openshift     2.3,2.4,2.5,latest
> > ```
> > And service catalog is not enabled in the 4.0 cluster causes blocking of `test-service-e2e` `test-watch-e2e` and `test-link-e2e`. I am not sure how service catalog and wildfly component types will be the part of prow setup.
> > This is something we will have to look into to see if we can get openshift ci black magic to help us.
> 
> If not, we should keep steps handy for how these can be added once a cluster is up
Right, I am looking for better solution. Will update once find something promising. 
> 
> > ```
> > $ odo catalog list services
> >  ✗  unable to list services because Service Catalog is not enabled in your cluster: unable to get cluster serviceClassExternalName: unable to get cluster service classes: unable to list cluster service classes: clusterserviceclasses.servicecatalog.k8s.io is forbidden: User "developer" cannot list resource "clusterserviceclasses" in API group "servicecatalog.k8s.io" at the cluster scope
> > ```
> > @kadel @mohammedzee1000 Please share your thoughts

**Comment by kadel** [comment]
I have noticed that there is a TemplateServiceBroker in OpenShift 4.0 operator hub.  But so far I was not able to make it work

**Comment by amitkrout** [comment]
> I have noticed that there is a TemplateServiceBroker in OpenShift 4.0 operator hub. But so far I was not able to make it work

Ahh, No issue. I can give a try. Can you please share the link for OpenShift 4.0 operator hub.

**Comment by kadel** [comment]
> Ahh, No issue. I can give a try. Can you please share the link for OpenShift 4.0 operator hub.

it is inside the OpenShift 4.0 webconsole

**Comment by mohammedzee1000** [comment]
> > I have noticed that there is a TemplateServiceBroker in OpenShift 4.0 operator hub. But so far I was not able to make it work
> 
> Ahh, No issue. I can give a try. Can you please share the link for OpenShift 4.0 operator hub.

We will need to figure way to do from the command line though

**Comment by amitkrout** [comment]
@mohammedzee1000 Java test was failing due to the output format fetched by terminal specific command like ```grep```. I run the same ```make test-java-e2e``` on my pr #1410 and the good news is that all the scenario goes fine. 

After merging of  pr #1410 into master then only we can add the test-java-e2e in OpenShift CI cc_ @kadel

**Comment by amitkrout** [comment]
```e2e_test``` - Block on #1454
```component_test``` - Need to verify some failure
```java_test``` - Block on #1454
```link_test``` - Block on service catalog 
```service_test``` - Block on service catalog 
```source_test``` - Tag <TargetFramework> - https://github.com/redhat-developer/odo/blob/master/tests/e2e/examples/source/dotnet/asp-net-hello-world.csproj#L4 need to be updated to 2.2 in a separate pr for prow setup to avoid failure like.
```
/opt/rh/rh-dotnet22/root/usr/lib64/dotnet/sdk/2.2.104/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(208,5): error NETSDK1005: Assets file '/opt/app-root/src/obj/project.assets.json' doesn't have a target for '.NETCoreApp,Version=v2.2'. Ensure that restore has run and that you have included 'netcoreapp2.2' in the TargetFrameworks for your project. [/opt/app-root/src/asp-net-hello-world.csproj]
```
Note: ```TargetFramework - 2.2``` is not supported on OpenShift 3.11.0
```watch_test``` - Block on service catalog 

To get a breakthrough on service catalog, i shoot a question in https://coreos.slack.com/archives/CEKNRGF25/p1552459439684900. Unfortunately no replay on my question.

@kadel @mohammedzee1000 Please share If you guys find any solution of enabling service catalog on OpenShift 4.0 cluster.

**Comment by amitkrout** [comment]
> > > I have noticed that there is a TemplateServiceBroker in OpenShift 4.0 operator hub. But so far I was not able to make it work
> > 
> > 
> > Ahh, No issue. I can give a try. Can you please share the link for OpenShift 4.0 operator hub.
> 
> We will need to figure way to do from the command line though

I installed ```Automation Broker Operator``` and ```Template Service Broker Operator``` In the default name space through the UI console. No luck, I am gettingg the same error 

I think something else is needed to enable service catalog

**Comment by mohammedzee1000** [comment]
> > > > I have noticed that there is a TemplateServiceBroker in OpenShift 4.0 operator hub. But so far I was not able to make it work
> > > 
> > > 
> > > Ahh, No issue. I can give a try. Can you please share the link for OpenShift 4.0 operator hub.
> > 
> > 
> > We will need to figure way to do from the command line though
> 
> I installed `Automation Broker Operator` and `Template Service Broker Operator` In the default name space through the UI console. No luck, I am gettingg the same error
> 
> I think something else is needed to enable service catalog

Ok for now, lets get the rest of the tests in. We will think about the tests needed service catalog lat. Still continue research on service catalog as well
